File: UniserCommerce/Entity/Concrete/Product.cs
﻿using Entity.Abstract;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entity.Concrete
{
    public class Product : ITable
    {
        public int Id { get; set; }
        public string ProductName { get; set; }
        public int Quantity { get; set; }
        public bool InStock { get; set; }
        public double Price { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }
    }
}

File: UniserCommerce/UniserCommerce.UI/wwwroot/js/AdminJs/Product.js
$(document).ready(function () {
    FillCategoryList();

    debugger;


    function AddNewProduct() {


        var product = {
            ProductName: $("#productName").val(),
            Quantity: $("#quantity").val(),
            Price: $("#price").val(),
            CategoryId: $("#categorySelect").val()
        }

        $.ajax({
            type: "POST",
            url: "/Admin/Admin/AddProduct",
            dataType: "json",
            contentType: "application/json; charset=utf-8",
            data: JSON.stringify(product),
            success: function (response) {

                console.log(response)
            },
            error: function () {
                alert("An error occurred while adding the category.");
            }
        })
    }
    
  
})


console.log($("#test").val());



function FillCategoryList() {
    let selectBox = $("#categorySelect");

    $.ajax({
        type: "GET",
        url: "/Admin/Admin/GetCategories",
        dataType: "json",
        success: function (data) {
            for (let i = 0; i < data.length; i++) {
                selectBox.append($('<option>',
                    {
                        value: data[i].Id,
                        text: data[i].Name
                    }
                ));
            }
        }
    })
}



